metrics/prometheus_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2020 NTT Innovation Institute, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"time"
+
+	"github.com/cloudwan/gohan/util"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Prometheus metrics", func() {
+	var (
+		pe          *prometheusExporter
+		emptyConfig *util.Config
+	)
+
+	BeforeEach(func() {
+		pe = &prometheusExporter{}
+		emptyConfig = util.NewConfig(map[string]interface{}{})
+	})
+
+	It("Has reasonable defaults after setup", func() {
+		Expect(pe.Setup(emptyConfig, ":1234")).To(Succeed())
+
+		Expect(pe.config.flushInterval).To(Equal(10 * time.Second))
```

Comment:
I've seen this being repeated later, why not just `Expect(pe.config).To(Equal())` or `thenConfigIs(pe.config, ...)?`

Reply:
There is a small catch. This test case does not check the value of serverAddress as I have other test cases dedicated for that, because there is some additional logic behind it.

If you really want, I can merge them.